test/phoenix/router/pipe_through_deprecation_test.exs

```diff
@@ -0,0 +1,191 @@
+defmodule Phoenix.Router.PipeThroughDeprecationTest do
```

Comment:
We should not have a new test file only for this and the tests can definitely be shorter. We only need to check for the warning, no need to for the other success cases.